drivers/timer/nrf_rtc_timer.c

```diff
@@ -462,13 +480,26 @@ void rtc1_nrf5_isr(void *arg)
 #endif
 
 #ifdef CONFIG_TICKLESS_KERNEL
+	if (!expected_sys_ticks) {
+		if (_sys_clock_always_on) {
+			_sys_clock_tick_count = _get_elapsed_clock_time();
+			program_max_cycles();
+		}
+		return;
+	}
 	_sys_idle_elapsed_ticks = expected_sys_ticks;
 	/* Initialize expected sys tick,
 	 * It will be later updated based on next timeout.
 	 */
 	expected_sys_ticks = 0;
 	/* Anounce elapsed of _sys_idle_elapsed_ticks systicks*/
 	_sys_clock_tick_announce();
+
+	/* _sys_clock_tick_announce() could cause new programming */
+	if (!expected_sys_ticks && _sys_clock_always_on) {
+		_sys_clock_tick_count = _get_elapsed_clock_time();
```

Comment:
_sys_clock_tick_count is already updated in program_max_cycles().

Reply:
will fix it